src/linkgraph/mcf.cpp

```diff
@@ -284,12 +284,19 @@ void MultiCommodityFlow::Dijkstra(NodeID source_node, PathVector &paths)
 				capacity /= 100;
 				if (capacity == 0) capacity = 1;
 			}
-			/* punish in-between stops a little */
+			/* Prioritize the fastest route for passengers and express cargo,
+			 * and the shortest route for other classes of cargo.
+			 * In-between stops are punished with a 1 tile or 1 day penalty. */
 			uint distance = DistanceMaxPlusManhattan(this->job[from].XY(), this->job[to].XY()) + 1;
+			/* Compute a default travel time from the distance and an average speed of 1 tile/day. */
+			uint time = (edge.TravelTime() != 0) ? edge.TravelTime() + DAY_TICKS : distance * DAY_TICKS;
+			uint distance_anno = (IsCargoInClass(this->job.Cargo(), CC_PASSENGERS) ||
+				IsCargoInClass(this->job.Cargo(), CC_EXPRESS)) ? time : distance;
```

Comment:
I'd include `CC_MAIL` for time as well.

Reply:
This makes sense, I added the CC_MAIL class.

I wonder if mail, passengers, and maybe fruits could be added to the CC_EXPRESS class? A comment in cargo_type.h states that CC_EXPRESS is "also possible for passengers", and it would simplify this time/distance logic.

Reply:
No, changing the pre-defined default cargoes is a bad idea as a heap of NewGRFs depend on it.

Reply:
Comment above needs updating too

Reply:
Probably best to pull out the ternary condition into its own variable too

Reply:
I've updated the forgotten comment and moved the ternary.